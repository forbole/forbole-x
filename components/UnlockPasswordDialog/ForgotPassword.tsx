import { Box, Button, DialogActions, DialogContent, DialogContentText } from '@material-ui/core';
import useTranslation from 'next-translate/useTranslation';
import React from 'react';
import useStyles from './styles';

interface ForgotPasswordProps {
  onReset(): void;
}

const ForgotPassword: React.FC<ForgotPasswordProps> = ({ onReset }) => {
  const { t } = useTranslation('common');
  const classes = useStyles();

  return (
    <>
      <DialogContent>
        <DialogContentText>{t('forgot password description')}</DialogContentText>
      </DialogContent>
      <DialogActions className={classes.action}>
        <Button
          className={classes.button}
          variant="contained"
          color="primary"
          onClick={() => onReset()}>
          {t('reset')}
        </Button>
        <Box my={3.5} />
      </DialogActions>
    </>
  );
};

export default ForgotPassword;
