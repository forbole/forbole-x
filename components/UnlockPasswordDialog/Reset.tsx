import { Button, DialogActions, DialogContent, DialogContentText } from '@material-ui/core';
import useTranslation from 'next-translate/useTranslation';
import React from 'react';
import useStyles from './styles';

interface ResetProps {
  onCancel(): void;
  onResetApp(): void;
}

const Reset: React.FC<ResetProps> = ({ onCancel, onResetApp }) => {
  const { t } = useTranslation('common');
  const classes = useStyles();

  return (
    <>
      <DialogContent>
        <DialogContentText>{t('reset description')}</DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button
          className={classes.resetButton}
          variant="contained"
          color="secondary"
          onClick={() => onCancel()}>
          {t('cancel')}
        </Button>
        <Button
          className={classes.resetButton}
          variant="contained"
          color="primary"
          onClick={() => onResetApp()}>
          {t('reset')}
        </Button>
      </DialogActions>
    </>
  );
};

export default Reset;
