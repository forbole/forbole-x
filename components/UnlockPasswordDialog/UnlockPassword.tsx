import { Button, DialogActions, DialogContent, DialogContentText } from '@material-ui/core'
import useTranslation from 'next-translate/useTranslation'
import React from 'react'
import useStyles from './styles'
import PasswordInput from '../PasswordInput'
import { useWalletsContext } from '../../contexts/WalletsContext'

interface UnlockPasswordProps {
  onForgot: () => void
}

const UnlockPassword: React.FC<UnlockPasswordProps> = ({ onForgot }) => {
  const { t } = useTranslation('common')
  const classes = useStyles()
  const [password, setPassword] = React.useState('')
  const { unlockWallets } = useWalletsContext()
  const [error, setError] = React.useState('')

  const onButtonClick = React.useCallback(async () => {
    try {
      setError('')
      await unlockWallets(password)
    } catch (err) {
      setError(t(err.message))
      setPassword('')
    }
  }, [password, setError, setPassword])

  return (
    <form
      noValidate
      className={classes.action}
      onSubmit={(e) => {
        e.preventDefault()
        onButtonClick()
      }}
    >
      <DialogContent>
        <DialogContentText>{t('unlock password description')}</DialogContentText>
        <PasswordInput
          placeholder={t('password')}
          value={password}
          error={!!error}
          helperText={error}
          onChange={(e) => setPassword(e.target.value)}
        />
      </DialogContent>
      <DialogActions className={classes.action}>
        <Button
          disabled={password.length < 6}
          className={classes.button}
          variant="contained"
          color="primary"
          type="submit"
        >
          {t('next')}
        </Button>
        <Button className={classes.forgotButton} onClick={() => onForgot()}>
          {t('forgot password?')}
        </Button>
      </DialogActions>
    </form>
  )
}

export default UnlockPassword
