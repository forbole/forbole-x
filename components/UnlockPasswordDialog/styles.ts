import { makeStyles } from '@material-ui/core/styles'
import { CustomTheme } from '../../misc/theme'

const useStyles = makeStyles(
  (theme: CustomTheme) => ({
    passwordRequirement: {
      marginTop: theme.spacing(1),
    },
    button: {
      margin: theme.spacing(1, 3),
      marginTop: theme.spacing(20),
    },
    dialog: {
      background: theme.palette.dialogBackground,
    },
    resetButton: {
      flex: 1,
      margin: theme.spacing(4, 2),
      marginTop: theme.spacing(6),
      color: 'white',
    },
    forgotButton: {
      marginBottom: theme.spacing(2),
      color: theme.palette.primary.main,
      width: 'fit-content',
      alignSelf: 'center',
    },
    action: {
      display: 'contents',
    },
    closeButton: {
      position: 'absolute',
      top: theme.spacing(2),
      right: theme.spacing(2),
    },
    backButton: {
      position: 'absolute',
      top: theme.spacing(2),
      left: theme.spacing(2),
    },
  }),
  {
    name: 'HookGlobalStyles',
    index: 2,
  }
)

export default useStyles
