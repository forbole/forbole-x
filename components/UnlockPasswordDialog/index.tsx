import { Dialog, DialogTitle, IconButton } from '@material-ui/core'
import useTranslation from 'next-translate/useTranslation'
import React from 'react'
import { useRouter } from 'next/router'
import useStyles from './styles'
import BackIcon from '../../assets/images/icons/icon_back.svg'
import { useWalletsContext } from '../../contexts/WalletsContext'
import ForgotPassword from './ForgotPassword'
import UnlockPassword from './UnlockPassword'
import Reset from './Reset'
import useIconProps from '../../misc/useIconProps'
import useStateHistory from '../../misc/useStateHistory'
import useIsMobile from '../../misc/useIsMobile'

enum UnlockPasswordStage {
  UnlockPasswordStage = 'unlock',
  ForgotPasswordStage = 'forgot',
  ResetStage = 'reset',
}

interface Content {
  title: string
  content: React.ReactNode
  dialogWidth?: 'xs' | 'sm' | 'md' | 'lg' | 'xl'
}

const UnlockPasswordDialog: React.FC = () => {
  const { t } = useTranslation('common')
  const classes = useStyles()
  const iconProps = useIconProps()
  const { wallets, reset, unlockWallets } = useWalletsContext()
  const isMobile = useIsMobile()
  const {
    query: { password: defaultPassword },
  } = useRouter()
  const [isReset, setReset] = React.useState(false)
  const [stage, setStage, toPrevStage, isPrevStageAvailable] = useStateHistory<UnlockPasswordStage>(
    UnlockPasswordStage.UnlockPasswordStage
  )

  const forgotPassword = React.useCallback(() => {
    setStage(UnlockPasswordStage.ForgotPasswordStage)
  }, [setStage])

  const resetAll = React.useCallback(() => {
    setStage(UnlockPasswordStage.ResetStage)
  }, [setStage])

  const cancel = React.useCallback(() => {
    setStage(UnlockPasswordStage.UnlockPasswordStage)
  }, [setStage])

  const resetApp = React.useCallback(async () => {
    await reset()
    setReset(true)
  }, [reset, wallets])

  const content: Content = React.useMemo(() => {
    switch (stage) {
      case UnlockPasswordStage.ResetStage:
        return {
          content: <Reset onCancel={cancel} onResetApp={resetApp} />,
          title: t('reset'),
        }
      case UnlockPasswordStage.ForgotPasswordStage:
        return {
          content: <ForgotPassword onReset={resetAll} />,
          title: t('forgot password'),
        }
      case UnlockPasswordStage.UnlockPasswordStage:
      default:
        return {
          content: <UnlockPassword onForgot={forgotPassword} />,
          title: t('unlock password title'),
        }
    }
  }, [stage, t])

  React.useEffect(() => {
    if (defaultPassword) {
      unlockWallets(String(defaultPassword))
    }
  }, [defaultPassword, unlockWallets])

  return (
    <Dialog fullWidth open={!(wallets.length !== 0 || isReset)} fullScreen={isMobile}>
      {isPrevStageAvailable && stage !== 'unlock' ? (
        <IconButton className={classes.backButton} onClick={toPrevStage}>
          <BackIcon {...iconProps} />
        </IconButton>
      ) : null}
      {content.title ? <DialogTitle>{content.title}</DialogTitle> : null}
      {content.content}
    </Dialog>
  )
}

export default UnlockPasswordDialog
