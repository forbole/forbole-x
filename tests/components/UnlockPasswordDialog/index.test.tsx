import React from 'react';
import renderer from 'react-test-renderer';
import UnlockPasswordDialog from '../../../components/UnlockPasswordDialog';

const mockWalletsContext = {
  password: '',
  unlockWallets: jest.fn().mockResolvedValue('done'),
  wallets: [],
};
jest.mock('next/router', () => ({
  useRouter: () => ({
    query: {},
  }),
}));
jest.mock('@material-ui/core/Dialog', () => props => <div id="dialog" {...props} />);
jest.mock('../../../components/PasswordInput', () => props => (
  <div id="PasswordInput" {...props} />
));

jest.mock('../../../contexts/WalletsContext', () => ({
  useWalletsContext: () => mockWalletsContext,
}));

describe('component: UnlockPasswordDialog', () => {
  it('renders correctly', () => {
    const component = renderer.create(<UnlockPasswordDialog />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders non empty wallets correctly', () => {
    mockWalletsContext.wallets = ['wallet'];
    const component = renderer.create(<UnlockPasswordDialog />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders non empty password correctly', () => {
    mockWalletsContext.password = 'password';
    const component = renderer.create(<UnlockPasswordDialog />);
    renderer.act(() => undefined);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly when password is changed', () => {
    const component = renderer.create(<UnlockPasswordDialog />);
    renderer.act(() => {
      component.root
        .findByProps({
          id: 'PasswordInput',
        })
        .props.onChange({
          target: {
            value: 'password',
          },
        });
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('calls unlockWallets when button is clicked', async () => {
    const component = renderer.create(<UnlockPasswordDialog />);
    renderer.act(() => {
      component.root
        .findByProps({
          id: 'PasswordInput',
        })
        .props.onChange({
          target: {
            value: 'password',
          },
        });
    });
    await renderer.act(async () => {
      await component.root.findByType('form').props.onSubmit({ preventDefault: jest.fn() });
    });
    expect(mockWalletsContext.unlockWallets).toBeCalledWith('password');
  });
  it('renders error state correctly', async () => {
    mockWalletsContext.unlockWallets.mockRejectedValueOnce({ message: 'invalid password' });
    const component = renderer.create(<UnlockPasswordDialog />);
    renderer.act(() => {
      component.root
        .findByProps({
          id: 'PasswordInput',
        })
        .props.onChange({
          target: {
            value: 'password',
          },
        });
    });
    await renderer.act(async () => {
      await component.root.findByType('form').props.onSubmit({ preventDefault: jest.fn() });
    });
    expect(mockWalletsContext.unlockWallets).toBeCalledWith('password');
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

afterEach(() => {
  mockWalletsContext.password = '';
  mockWalletsContext.wallets = [];
  jest.clearAllMocks();
});
