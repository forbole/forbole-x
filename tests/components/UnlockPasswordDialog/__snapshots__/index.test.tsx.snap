// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: UnlockPasswordDialog renders correctly 1`] = `
<div
  PaperProps={
    Object {
      "className": "HookGlobalStyles-dialog-3",
    }
  }
  fullScreen={false}
  fullWidth={true}
  id="dialog"
  open={true}
>
  <div
    className="MuiDialogTitle-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h6"
    >
      common:unlock password title
    </h2>
  </div>
  <form
    className="HookGlobalStyles-action-6"
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="MuiDialogContent-root"
    >
      <p
        className="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        common:unlock password description
      </p>
      <div
        error={false}
        helperText=""
        id="PasswordInput"
        onChange={[Function]}
        placeholder="common:password"
        value=""
      />
    </div>
    <div
      className="MuiDialogActions-root HookGlobalStyles-action-6 MuiDialogActions-spacing"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-2 MuiButton-containedPrimary Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          common:next
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text HookGlobalStyles-forgotButton-5"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          common:forgot password?
        </span>
      </button>
    </div>
  </form>
</div>
`;

exports[`component: UnlockPasswordDialog renders correctly when password is changed 1`] = `
<div
  PaperProps={
    Object {
      "className": "HookGlobalStyles-dialog-3",
    }
  }
  fullScreen={false}
  fullWidth={true}
  id="dialog"
  open={true}
>
  <div
    className="MuiDialogTitle-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h6"
    >
      common:unlock password title
    </h2>
  </div>
  <form
    className="HookGlobalStyles-action-6"
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="MuiDialogContent-root"
    >
      <p
        className="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        common:unlock password description
      </p>
      <div
        error={false}
        helperText=""
        id="PasswordInput"
        onChange={[Function]}
        placeholder="common:password"
        value="password"
      />
    </div>
    <div
      className="MuiDialogActions-root HookGlobalStyles-action-6 MuiDialogActions-spacing"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-2 MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          common:next
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text HookGlobalStyles-forgotButton-5"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          common:forgot password?
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</div>
`;

exports[`component: UnlockPasswordDialog renders error state correctly 1`] = `
<div
  PaperProps={
    Object {
      "className": "HookGlobalStyles-dialog-3",
    }
  }
  fullScreen={false}
  fullWidth={true}
  id="dialog"
  open={true}
>
  <div
    className="MuiDialogTitle-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h6"
    >
      common:unlock password title
    </h2>
  </div>
  <form
    className="HookGlobalStyles-action-6"
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="MuiDialogContent-root"
    >
      <p
        className="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        common:unlock password description
      </p>
      <div
        error={true}
        helperText="common:invalid password"
        id="PasswordInput"
        onChange={[Function]}
        placeholder="common:password"
        value=""
      />
    </div>
    <div
      className="MuiDialogActions-root HookGlobalStyles-action-6 MuiDialogActions-spacing"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-2 MuiButton-containedPrimary Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          common:next
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text HookGlobalStyles-forgotButton-5"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          common:forgot password?
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</div>
`;

exports[`component: UnlockPasswordDialog renders non empty password correctly 1`] = `
<div
  PaperProps={
    Object {
      "className": "HookGlobalStyles-dialog-3",
    }
  }
  fullScreen={false}
  fullWidth={true}
  id="dialog"
  open={true}
>
  <div
    className="MuiDialogTitle-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h6"
    >
      common:unlock password title
    </h2>
  </div>
  <form
    className="HookGlobalStyles-action-6"
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="MuiDialogContent-root"
    >
      <p
        className="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        common:unlock password description
      </p>
      <div
        error={false}
        helperText=""
        id="PasswordInput"
        onChange={[Function]}
        placeholder="common:password"
        value=""
      />
    </div>
    <div
      className="MuiDialogActions-root HookGlobalStyles-action-6 MuiDialogActions-spacing"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-2 MuiButton-containedPrimary Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          common:next
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text HookGlobalStyles-forgotButton-5"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          common:forgot password?
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</div>
`;

exports[`component: UnlockPasswordDialog renders non empty wallets correctly 1`] = `
<div
  PaperProps={
    Object {
      "className": "HookGlobalStyles-dialog-3",
    }
  }
  fullScreen={false}
  fullWidth={true}
  id="dialog"
  open={false}
>
  <div
    className="MuiDialogTitle-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h6"
    >
      common:unlock password title
    </h2>
  </div>
  <form
    className="HookGlobalStyles-action-6"
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="MuiDialogContent-root"
    >
      <p
        className="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        common:unlock password description
      </p>
      <div
        error={false}
        helperText=""
        id="PasswordInput"
        onChange={[Function]}
        placeholder="common:password"
        value=""
      />
    </div>
    <div
      className="MuiDialogActions-root HookGlobalStyles-action-6 MuiDialogActions-spacing"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-2 MuiButton-containedPrimary Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          common:next
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text HookGlobalStyles-forgotButton-5"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          common:forgot password?
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</div>
`;
